### ISSUE TITLE
第3,4回の内容

### ISSUE THREAD

**Issue by anryu15** [shun74/pytorch-training#3]
## 第３,4回の内容の相談

- 何を教えるか

**Comment by shun74** [comment]
@Kosuke-Sugimoto @anryu15 
以下の内容から具体的な内容にしていきましょう
## PyTorchのかんたんな導入＆モデルの作り方

* PyTorchはどのようなフレームワークなのか
* 深層学習のモデルに関する理論
* PyTorchでnn.Moduleを使ったモデルの作り方
* ConvやLinearやBatchNormを使った少し複雑なモデルの作り方

**Comment by shun74** [comment]
@Kosuke-Sugimoto 
このissueに合わせてスライド作成を行ってもらえると助かります。

**Comment by Kosuke-Sugimoto** [comment]
@shun74 @anryu15  
スライドを作ってみないと何とも言えませんが、今のところ、  
  
## 第三回  
- PyTorchはどのようなフレームワークなのか
- 深層学習のモデルに関する理論
  - Conv, BatchNorm, ReLU, Linear あたりの軽い説明
  - ↑ は第四回で使うものと対応付ける

## 第四回
- PyTorch で nn.Module を使ったモデルの作り方
  - forward, __init__ の定義方法
  - 恒等写像くらいのレベル感
- 第三回で取り扱ったモジュールを用いて少し複雑なモデルの作成

くらいの規模感で考えています

**Comment by shun74** [comment]
@Kosuke-Sugimoto 
PyTorchの基礎に触れる回をそういえば設けていなかったので、第3回目のほとんどの時間は機械学習モデルの説明とPyTorchの簡単な説明に使ってしまってもいいかもしれないです。

モデル作成はそこまで守備範囲が広くないので次のようなベースで、少し案を再構成してもらってもいいですか？
* 第三回：深層学習理論の簡単な説明＆PyTorchの機能に関する説明(モデル、データローダー、トレーニングなどに軽く触れる)。
* 第四回：PyTorchでのモデル作成。足し算をするだけのModule作成から、Conv, ReLU, BN使うところまで。

**Comment by Kosuke-Sugimoto** [comment]
@shun74  
了解しました。詳細はそれぞれ後で説明すると思うので省きますが、触り程度で説明を入れておきます

**Comment by Kosuke-Sugimoto** [comment]
@shun74 @anryu15 
PyTorch の初期化についてのスライドを入れるか否か
→ torch.init について
→ 知らなくてもそこまで問題はなさそう

**Comment by anryu15** [comment]
@Kosuke-Sugimoto 

知らなくても問題なければなくてもいいと思う。
資料作成して説明する上で時間が余計に余る場合とかは入れるぐらいでも！

**Comment by Kosuke-Sugimoto** [comment]
第三回の Google Slide です
https://docs.google.com/presentation/d/1GcPBZ56-o7H_uAw9s_B7_DC7tPQ3MTp1Eh0yM_xZmpU/edit?usp=sharing

**Comment by Kosuke-Sugimoto** [comment]
https://docs.google.com/presentation/d/1GcPBZ56-o7H_uAw9s_B7_DC7tPQ3MTp1Eh0yM_xZmpU/edit?usp=sharing

**Comment by shun74** [comment]
> 第三回の Google Slide です https://docs.google.com/presentation/d/1GcPBZ56-o7H_uAw9s_B7_DC7tPQ3MTp1Eh0yM_xZmpU/edit?usp=sharing

@Kosuke-Sugimoto 
スライドの内容は今の所大丈夫だと思います。スクリプトを含め全体が完成したらPull Requestにしてください。
一つスライドの書き方に関してですが、ほとんどのスライドが黒字のみになっているので強調したい部分を赤字や太字にする方が見やすくなると思います。

**Comment by Kosuke-Sugimoto** [comment]
第四回の Google Slide です
https://docs.google.com/presentation/d/1h9Hf1K0J1tNeJ0Pd41ueQifQP3AavrYqnL7LMEqTj08/edit?usp=sharing
作成途中ですが、何かあったらコメントください

**Comment by Kosuke-Sugimoto** [comment]
@shun74 @anryu15 @hdhono 
第四回のスライドの説明部分（演習を除いた部分）が概ね完成したので、コメントお願いします
演習部分に関しては近日中にPRとともに追加します
リンクについては上のものと共通です
